### PR TITLE
New boilerplate

### PR DESCRIPTION
**Pull request by ipekt**
Move grunt boilerplate to grunt folder
Add webpack, react, redux project boilerplate

### CODE REVIEW

**Comment by scott-christopher** [comment]
Is there any benefit in keeping the grunt stuff around or can we just scrap it?

**Comment by ipekt** [comment]
I woudnt use grunt. @finderau/front-end

File: webpack/styles/components/_heading.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+h1 {
```
Comment (darylljann) [inline webpack/styles/components/_heading.scss]: Curious what's this for?

Comment (ipekt) [inline webpack/styles/components/_heading.scss]: Just wanted to add folder structure for SASS and test that webpack compiles SASS correctly.

File: webpack/.sass-lint.yml [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    - 2
+    -
+      leading-underscore: false
+      filename-extension: false
+  class-name-format:
+    - 0
+    -
+      allow-leading-underscore: false
+      convention: ^([a-zA-Z]*)-(([a-zA-Z]*)+)(__(([a-zA-Z0-9]+)+))?(--(([a-zA-Z0-9])+)+)?$
+      convention-explanation: 'Class name does not follow conventions: .luna-[className][__descendantName][--modifierName], .u-utilityName, or .is-stateName'
```
Comment (StrangePurple) [inline webpack/.sass-lint.yml]: Can ditch the mentions of luna- in here, and the prefix part of the convention above

Comment (ipekt) [inline webpack/.sass-lint.yml]: Do you mean remove `convention` setting from `class-name-format`?

File: webpack/styles/main.scss [hunk 3]
```diff
@@ -0,0 +1,8 @@
+@charset "UTF-8"; // sass-lint:disable-line quotes
+
+@import '../node_modules/grunt-finder-SASS/assets/stylesheets/finder_sass';
```
Comment (StrangePurple) [inline webpack/styles/main.scss]: Would be nice to not rely on this repo at all, do we know what we're still using from it?

File: webpack/styles/main.scss [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+@import '../node_modules/luna/src/toolkit/toolkit';
+@import '../node_modules/luna/src/components/typography';
+
+@import './constructs/project';
```
Comment (StrangePurple) [inline webpack/styles/main.scss]: (minor detail) Constructs should technically be loaded after components

File: webpack/.sass-lint.yml (hunk 2 above)
Comment (StrangePurple) [inline webpack/.sass-lint.yml]: yep, can remove the first part of it which is basically looking for a `luna-` (a-zA-Z)-
That way the convention becomes blockName__elementName--modifierName only

File: webpack/styles/main.scss (hunk 3 above)
Comment (ipekt) [inline webpack/styles/main.scss]: Seems like breakpoint mixin is the only one we require in new projects

File: webpack/package.json [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+  "description": "New Project",
+  "private": true,
+  "main": "js/index.js",
+  "scripts": {
+    "build": "NODE_ENV=production webpack -p --progress",
+    "watch": "NODE_ENV=development webpack -d --progress --watch",
+    "build:dev": "NODE_ENV=development webpack -d --progress",
+    "clean": "rm -rf ./build && mkdir ./build",
+    "start": "NODE_ENV=development webpack-dev-server -d --open",
+    "test": "npm run lint && npm run test:unit",
```
Comment (kopacki) [inline webpack/package.json]: I would suggest having `lint` task called as `pretest`.

File: webpack/package.json [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    "ramda": "~0.23.0",
+    "react": "~15.4.1",
+    "react-dom": "~15.4.1",
+    "react-redux": "~4.4.5",
+    "redux": "~3.6.0",
+    "redux-mock-store": "~1.2.1",
+    "redux-thunk": "~2.1.0",
+    "sinon": "~1.17.5",
+    "whatwg-fetch": "~1.0.0"
+  }
```
Comment (kopacki) [inline webpack/package.json]: Can we use `yarn` as a dependency manager?

File: webpack/styles/main.scss [hunk 7]
```diff
@@ -0,0 +1,8 @@
+@charset "UTF-8"; // sass-lint:disable-line quotes
+
+@import '../node_modules/grunt-finder-SASS/assets/stylesheets/finder_sass';
+@import '../node_modules/luna/src/toolkit/toolkit';
```
Comment (kopacki) [inline webpack/styles/main.scss]: Try `~luna/src/toolkit/toolkit`instead of the current form.

File: webpack/package.json (hunk 6 above)
Comment (scott-christopher) [inline webpack/package.json]: Can yarn handle git+ssh dependencies yet?

File: webpack/styles/main.scss (hunk 7 above)
Comment (ipekt) [inline webpack/styles/main.scss]: Nice

**Comment by kopacki** [comment]
What do you think to have project boilerplate per branch, so we have particular branch for the setup and avoid nesting ;)

**Comment by GemmaStiles** [comment]
@ipekt What is needed to get some movement on this PR?

**Comment by ipekt** [comment]
@GemmaStiles I have removed grunt boilerplate following the comments above and add readme. Since this is the only boilerplate in this repo, I dont think we need to put it in seperate branch. This PR needs to be approved to move forward.

**Pull request merged**